### PR TITLE
feat(file-processors): Add an option to serialize Docling tables as Markdown

### LINKED ISSUE #6403: Serialize tables inside Docling chunks as Markdown
Docling currently serializes tables inside OGX chunks as row triplets. This loses the table layout and makes the retrieved text harder for people and models to understand.

I would like OGX to add an optional `chunking_use_markdown_tables` setting to the `remote::docling-serve` file processor.

When the setting is `true`, tables inside chunks should use normal Markdown table syntax. When it is missing or `false`, the current row-triplet format should stay the same.

This would preserve table headers and rows in the retrieved text. It should make table content easier to use for RAG and more useful when creating precise citations.

The implementation can use Docling's normal convert endpoint with chunk output and custom hybrid chunking options. It supports both the sync and async OGX paths. Tests cover both paths, and the async path was also tested end to end with Docling Serve v1.30.0 and pgvector.

This requires a Docling client and Docling Serve version that support chunk output through the convert endpoint. The client needs `docling-slim` 2.117.0 or newer. Docling Serve v1.30.0 is the version tested so far.

One design question is how OGX should handle older Docling Serve versions. My suggestion is to document the compatible version and return a clear error. Falling back to the old chunking endpoint does not seem useful because the Docling maintainers have said that it is about to be deprecated:
https://github.com/docling-project/docling/pull/3983

I have an implementation here:
https://github.com/pgustafs/ogx/tree/feat/docling-markdown-table-chunks

### PR DESCRIPTION
**Pull request by pgustafs**
# What does this PR do?

This PR lets callers ask the `remote::docling-serve` file processor to serialize tables inside chunks as Markdown tables instead of row triplets.

The option is set for each request:

```python
options={"use_markdown_tables": True}
```

The option works in both places that can start file processing:

- Direct File Processors API requests.
- Files attached through the Vector Stores API.

Vector-store attachment already uses the File Processors API internally. This PR adds an optional `options` field to `OpenAIAttachFileRequest` and forwards it to the internal `ProcessFileRequest`. This keeps ingestion behavior consistent across both API entry points.

The option is optional. Existing requests that do not send it keep the current behavior. The Docling Serve provider config stays limited to connection and operational settings.

The File Processors API uses multipart form data. This PR also makes its existing `options` field accept a JSON object from multipart HTTP requests. This lets deployed OGX servers pass provider-specific options into `ProcessFileRequest.options`.

Invalid values, such as the string `"true"`, return a validation error instead of silently enabling the option.

This PR also changes chunking to use Docling's normal convert endpoint with chunk output. The old chunking endpoint is planned for deprecation, as discussed in [docling-project/docling#3983](https://github.com/docling-project/docling/pull/3983).

The implementation:

- Supports both automatic and static OGX chunk sizes.
- Reads chunk output from the ZIP archive returned by Docling.
- Keeps headings and page numbers from Docling.
- Accepts Docling options on direct file-processing requests and vector-store file attachments.
- Returns a clear error when the Docling Serve version does not support chunk output through the convert endpoint.
- Updates the minimum `docling-slim` version to `2.117.0`.
- Regenerates the OpenAPI specifications.

Chunking requires a compatible Docling Serve version, such as v1.30.0.

Closes #6403

## Test Plan

Focused file processor, multipart request, vector-store router, and attachment propagation tests:

```bash
uv run pytest -q \
  tests/unit/core/jobs/test_file_processor_routes.py \
  tests/unit/providers/file_processor/test_docling_serve.py \
  tests/unit/providers/file_processor/test_docling_metadata.py \
  tests/unit/providers/utils/memory/test_openai_vector_store_mixin.py \
  tests/unit/core/routers/test_vector_stores_abac.py
```

Result:

```text
75 passed
```

All code, formatting, type, generated-schema, API compatibility, and API coverage pre-commit checks passed:

```bash
uv run pre-commit run --all-files
```

The complete GitHub unit test job passed on commit `acc30038c`:

```text
unit-tests (3.12): passed
```

The unit tests check that:

- Multipart JSON is available in `ProcessFileRequest.options`.
- Invalid multipart options are rejected.
- Missing `use_markdown_tables` keeps the default behavior.
- Boolean values are passed to Docling.
- Non-boolean values are rejected.
- `OpenAIAttachFileRequest.options` reaches the internal `ProcessFileRequest` unchanged.
- Existing attachment requests without `options` still use automatic chunking.

The script below verifies the new attachment path against a running OGX server with Docling Serve and pgvector:

```python
#!/usr/bin/env -S uv run --script
# /// script
# requires-python = ">=3.11"
# dependencies = [
#   "httpx>=0.27,<1",
#   "reportlab>=4.2,<5",
# ]
# ///

import io
import os
import re
import time

import httpx
from reportlab.lib import colors
from reportlab.lib.pagesizes import letter
from reportlab.platypus import SimpleDocTemplate, Table, TableStyle


def make_pdf() -> bytes:
    output = io.BytesIO()
    document = SimpleDocTemplate(output, pagesize=letter)
    table = Table(
        [
            ["Probe Code", "Probe Region", "Probe Units"],
            ["MDTABLE-ALPHA-7319", "North Harbor", "17"],
            ["MDTABLE-BETA-8426", "South Ridge", "29"],
        ]
    )
    table.setStyle(
        TableStyle(
            [
                ("GRID", (0, 0), (-1, -1), 1, colors.black),
                ("FONTNAME", (0, 0), (-1, 0), "Helvetica-Bold"),
            ]
        )
    )
    document.build([table])
    return output.getvalue()


base_url = os.environ.get("OGX_URL", "http://localhost:8321/v1").rstrip("/")
headers = {"Authorization": f"Bearer {os.environ.get('OGX_API_KEY', 'none')}"}
client = httpx.Client(base_url=base_url, headers=headers, timeout=300)
vector_store_id = None
file_id = None

try:
    vector_store = client.post(
        "/vector_stores",
        json={
            "name": "markdown-table-attachment-test",
            "provider_id": "pgvector",
            "embedding_model": "sentence-transformers/nomic-ai/nomic-embed-text-v1.5",
            "embedding_dimension": 768,
        },
    )
    vector_store.raise_for_status()
    vector_store_id = vector_store.json()["id"]

    upload = client.post(
        "/files",
        data={"purpose": "assistants"},
        files={"file": ("markdown-table-test.pdf", make_pdf(), "application/pdf")},
    )
    upload.raise_for_status()
    file_id = upload.json()["id"]

    attachment = client.post(
        f"/vector_stores/{vector_store_id}/files",
        json={
            "file_id": file_id,
            "options": {"use_markdown_tables": True},
            "chunking_strategy": {
                "type": "static",
                "static": {
                    "max_chunk_size_tokens": 256,
                    "chunk_overlap_tokens": 32,
                },
            },
        },
    )
    attachment.raise_for_status()
    attachment_body = attachment.json()

    while attachment_body["status"] == "in_progress":
        time.sleep(1)
        attachment = client.get(f"/vector_stores/{vector_store_id}/files/{file_id}")
        attachment.raise_for_status()
        attachment_body = attachment.json()

    assert attachment_body["status"] == "completed", attachment_body

    content = client.get(f"/vector_stores/{vector_store_id}/files/{file_id}/content")
    content.raise_for_status()
    text = "\n".join(item["text"] for item in content.json()["data"])

    assert "| Probe Code" in text
    assert "| MDTABLE-ALPHA-7319" in text
    assert re.search(r"\|\s*-{3,}", text)
    assert not re.search(r"Probe\s+(Code|Region|Units)\s*=", text)

    search = client.post(
        f"/vector_stores/{vector_store_id}/search",
        json={"query": "MDTABLE-ALPHA-7319 North Harbor 17"},
    )
    search.raise_for_status()
    search_text = "\n".join(
        item["text"]
        for result in search.json()["data"]
        for item in result["content"]
    )
    assert "MDTABLE-ALPHA-7319" in search_text

    print("PASS: vector-store attachment forwarded the Markdown-table option to Docling.")
finally:
    if vector_store_id:
        client.delete(f"/vector_stores/{vector_store_id}").raise_for_status()
    if file_id:
        client.delete(f"/files/{file_id}").raise_for_status()
    client.close()
```

Execution output from OGX commit `acc30038c`, Docling Slim `2.120.1`, Docling Serve `v1.30.0`, and `remote::pgvector`:

```text
Providers verified: remote::pgvector and remote::docling-serve
Created vector store: vs_c93cb311-ceb8-4042-8ab6-e6718084bc09 (provider=pgvector)
Uploaded table PDF: file-f2fa8599b08e49338159e70210783203
Vector-store attachment completed with options.use_markdown_tables=true
Retrieved 3 stored Docling chunk(s) from pgvector

| Probe Code         | Probe Region   |   Probe Units |
|--------------------|----------------|---------------|
| MDTABLE-ALPHA-7319 | North Harbor   |            17 |

pgvector search returned the uniquely marked table row
PASS: vector-store attachment forwarded the Markdown-table option to Docling.
```

### CODE REVIEW

**Comment by mattf** [comment]
@alinaryan ptal

File: src/ogx/providers/remote/file_processor/docling_serve/config.py [hunk 1]
```diff
@@ -32,6 +32,10 @@ class DoclingServeFileProcessorConfig(BaseModel):
         le=4096,
         description="Default chunk size in tokens when chunking_strategy type is 'auto'",
     )
+    chunking_use_markdown_tables: bool = Field(
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/docling_serve/config.py]: ProcessFileRequest for the file processors api already has an options: dict[str, Any] field for provider-specific parameters. use_markdown_tables should be read from request.options at call time rather than added to the docling serve provider config. 
The config should stay limited to connection/operational settings (base_url, api_key, mode, default_chunk_size_tokens) — not Docling feature flags. This keeps OGX provider-agnostic and lets callers toggle per-request.

File: src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py [hunk 2]
```diff
@@ -281,19 +350,14 @@ async def _convert_and_chunk(
         chunking_strategy: VectorStoreChunkingStrategy,
         document_metadata: dict[str, Any],
     ) -> list[Chunk]:
-        """Convert and chunk a file via Docling Serve's hybrid chunker endpoint."""
-        url = f"{self.config.base_url}/v1/chunk/hybrid/file"
+        """Convert and chunk a file via Docling Serve's synchronous convert endpoint."""
+        url = f"{self.config.base_url}/v1/convert/file"
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py:381]: The convert endpoint with OutputFormat.CHUNKS + ZipTarget depends on [docling-project/docling#3983](https://github.com/docling-project/docling/pull/3983), which is still in draft. Let's wait for the upstream API to land before we build on it here? Otherwise we risk rework if the Docling side changes.

**Review by alinaryan — COMMENTED** [review]
Thanks for working on this! The code cleanup and the markdown tables feature are great improvements. A couple comments

File: src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py (hunk 2 above)
Comment (pgustafs) [reply src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py:381]: @alinaryan 

Thanks. I agree that OGX should not depend on a draft upstream API. I think the link to docling#3983 in the PR description caused some confusion.
 
This PR does not depend on production code from docling#3983. OutputFormat.CHUNKS, ConvertDocumentsOptions.chunking_options, HybridChunkerOptions, ZipTarget, and submit(..., target=ZipTarget()) are already available in the released Docling v2.117.0.

docling#3983 originally proposed a change to submit_chunk(), but that code was reverted. Its current diff only adds documentation, an example, and regression tests for the existing convert path.

I also tested this with the unmodified docling-slim v2.120.1 package and Docling Serve v1.30.0 in both sync and async modes.

I can update this PR description to make that clearer. Is there another unreleased part of the Docling API that you believe this PR uses?

File: src/ogx/providers/remote/file_processor/docling_serve/config.py (hunk 1 above)
Comment (pgustafs) [reply src/ogx/providers/remote/file_processor/docling_serve/config.py]: @alinaryan 
Thanks, I updated the PR.

`use_markdown_tables` is now read from `request.options` for each request and defaults to `false`. I removed it from the Docling Serve provider config. Both sync and async paths support it.

I also found that the multipart HTTP route rejected JSON values for the existing `options` field. I added generic JSON parsing so callers can send:

`{"use_markdown_tables": true}`

I added tests for the request option, invalid values, and both Docling paths.

File: src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py [hunk 3]
```diff
@@ -74,6 +78,83 @@ def _get_headers(self) -> dict[str, str]:
...
+            chunks.append(
+                Chunk(
+                    content=text,
+                    chunk_id=chunk_id,
+                    metadata=metadata,
+                    chunk_metadata=ChunkMetadata(
+                        chunk_id=chunk_id,
+                        document_id=document_id,
+                        source=document_metadata.get("filename", ""),
+                        content_token_count=raw_chunk.num_tokens or len(text.split()),
```
[the remark this answers is not in the document]
Comment (pgustafs) [reply src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py]: I will use an explicit is not None check and add a regression test for both 0 and None.

[the remark this answers is not in the document]
Comment (pgustafs) [reply src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py]: Fixed in 8b7646677. An explicit num_tokens value of 0 is now preserved, and only None uses the word-count fallback. I also added regression tests for both cases. The focused Docling Serve suite passes with 28 tests.

File: src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py (hunk 2 above)
Comment (alinaryan) [reply src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py:381]: Ahh you're right, I misread. I thought it was building on draft work, but if that PR is just docs/tests for already-shipped APIs, then we're good!

**Review by alinaryan — APPROVED** [review]

**Review by mattf — APPROVED** [review]
approved per @alinaryan
